--- git status ---
On branch joystick-tracking-trash
Your branch is up to date with 'origin/joystick-tracking-trash'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd
	modified:   source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
	modified:   source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	logs/rsl_rl/dodo_manage_joystick_tracking_ppo/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_14-18-24/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_14-40-21/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_14-55-25/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_15-19-10/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_15-24-27/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_15-28-25/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_15-34-00/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_15-50-51/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_15-55-17/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_16-02-52/
	logs/rsl_rl/dodo_velocity_tracking_ppo/2026-01-29_14-05-14/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt b/logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt
index bdde2a5..ab613f9 100644
Binary files a/logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt and b/logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt
index 6fd3518..c7e993b 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt
index 6e413b2..29c589c 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt
index 9520486..87f3f82 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt
index 098d581..615005b 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt
index 84a7351..fdda736 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt
index a4fd06f..b82bc1e 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt
index 343307d..b43e24c 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt
index 8a4485d..3e6a3f3 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt
index 30bce28..dd20a86 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt
index dc0e874..0017616 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt
index 3af0b31..88ac861 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt
index d16f272..6775546 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt
index d49b062..022cfc1 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt
index 4b6a6b4..1931741 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt
index c27f3d9..13d287b 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt
index 018ab73..d60cad2 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt
index 0efc4e8..57c6eba 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt
index 4c96b61..cc5b398 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt
index a235fde..3711c83 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt
index f7fe214..973c401 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt
index 36d574d..567f585 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt
index 5c2f48d..380578f 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt
index 189d172..f52b94b 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt
index 79380b9..3e92cf7 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt
index a9dbe7e..7e42eb8 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt
index e4e0ca5..0e44dab 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt
index 1c0a07a..5d58a5c 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt
index 12957f7..600e3be 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt
index 35099b6..4e7860a 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd
index a7adf67..838260c 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd
index 2697045..f7182a7 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd
index 2af19ea..dd8225f 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd
index 00e94a9..40c4936 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd
index ae3974d..3e63b9d 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
index 6ae6a3a..e0b28e0 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
@@ -95,7 +95,7 @@ class ActionsCfg:
     joint_pos = mdp.JointPositionActionCfg(
         asset_name="robot",
         joint_names=[".*"],
-        scale=0.5,          # 先用小一点，避免一上来动作过猛
+        scale=0.65,          # 先用小一点，避免一上来动作过猛
         use_default_offset=True,  # 以默认姿态为基准做偏移（如果你的版本支持）
     )
 
@@ -197,8 +197,8 @@ class EventCfg:
         func=mdp.reset_joints_by_offset,
         mode="reset",
         params={
-            "position_range": (-0.2, 0.2),
-            "velocity_range": (-0.1, 0.1),
+            "position_range": (-0.05, 0.05),
+            "velocity_range": (-0.05, 0.05),
         },
     )
 
@@ -233,15 +233,18 @@ class RewardsCfg:
         params={"command_name": "base_velocity", "std": 0.5},
     )
 
-    feet_air_time = RewTerm(
-        func=mdp.feet_air_time_positive_biped_snesor,
-        weight=0.3,
-        params={
-            "command_name": "base_velocity",
-            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
-            "threshold": 0.4,
-        },
-    )
+    yaw_rate = RewTerm(func=mdp.yaw_rate_l2, weight=-0.2)
+    lin_vel_y = RewTerm(func=mdp.lin_vel_y_l2, weight=-0.2)
+
+    # feet_air_time = RewTerm(
+    #     func=mdp.feet_air_time_positive_biped_snesor,
+    #     weight=0.0,
+    #     params={
+    #         "command_name": "base_velocity",
+    #         "sensor_cfg": SceneEntityCfg("contact_forces", body_names=["left_link_4", "right_link_4"]),
+    #         "threshold": 0.4,
+    #     },
+    # )
 
     action_rate = RewTerm(
         func=mdp.action_rate_l2,
@@ -250,22 +253,22 @@ class RewardsCfg:
         
     feet_slide = RewTerm(
         func=mdp.feet_slide,
-        weight=-2.0,
-        params={
-            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
-            "asset_cfg": SceneEntityCfg("robot", body_names=".*_link_4"),
-        },
-    )
-
-    single_support = RewTerm(
-        func=mdp.single_support_reward,
-        weight=0.3,   # 推荐 0.3 ~ 1.0
+        weight=-0.2,
         params={
             "sensor_cfg": SceneEntityCfg("contact_forces", body_names=["left_link_4", "right_link_4"]),
-            "force_threshold": 15.0,
+            "asset_cfg": SceneEntityCfg("robot", body_names=["left_link_4", "right_link_4"]),
         },
     )
 
+    # single_support = RewTerm(
+    #     func=mdp.single_support_reward,
+    #     weight=0.3,   # 推荐 0.3 ~ 1.0
+    #     params={
+    #         "sensor_cfg": SceneEntityCfg("contact_forces", body_names=["left_link_4", "right_link_4"]),
+    #         "force_threshold": 15.0,
+    #     },
+    # )
+
     alternate_steps = RewTerm(
         func=mdp.alternate_footstep_reward,
         weight=0.5,   # 从 0.2~1.0 试
@@ -288,7 +291,39 @@ class RewardsCfg:
         },
     )
 
+    feet_clearance = RewTerm(
+        func=mdp.feet_clearance_reward,
+        weight=0.25,
+        params={
+            "asset_cfg": SceneEntityCfg("robot", body_names=["left_link_4", "right_link_4"]),
+            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=["left_link_4", "right_link_4"]),
+            "target_height": 0.05,
+            "force_threshold": 15.0,
+        },
+    )
+    
+    #knee limit
+    knee_flex = RewTerm(
+        func=mdp.knee_flexion_target_exp,
+        weight=0.6,
+        params={
+            "asset_cfg": SceneEntityCfg("robot", joint_names=["left_joint_3", "right_joint_3"]),
+            "knee_target": 0.8,
+            "std": 0.25,
+        },
+    )   
 
+    swing_knee = RewTerm(
+        func=mdp.swing_knee_flexion_reward,
+        weight=0.5,
+        params={
+            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=["left_link_4", "right_link_4"]),
+            "knee_cfg": SceneEntityCfg("robot", joint_names=["left_joint_3", "right_joint_3"]),
+            "knee_target": 0.9,
+            "std": 0.25,
+            "force_threshold": 10.0,
+        },
+    )
 # --- 以下奖励项暂时不启用 ---
     # 髋关节移位置惩罚
     # hip_joint_move = RewTerm(func=mdp.hip_pos_manual_limit,
@@ -311,16 +346,7 @@ class RewardsCfg:
     #     params={"threshold": 0.98, "gear_ratio": {".*": 2.5}},
     # )
 
-    # feet_clearance = RewTerm(
-    #     func=mdp.feet_clearance_reward,
-    #     weight=0.25,
-    #     params={
-    #         "asset_cfg": SceneEntityCfg("robot", body_names=["left_link_4", "right_link_4"]),
-    #         "sensor_cfg": SceneEntityCfg("contact_forces", body_names=["left_link_4", "right_link_4"]),
-    #         "target_height": 0.05,
-    #         "force_threshold": 15.0,
-    #     },
-    # )
+
 
 
     # feet_flight = RewTerm(
diff --git a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py
index 3971ccb..d50c2fb 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py
@@ -214,8 +214,87 @@ def feet_lateral_separation_reward(
 
     return r * stance_any.float()
 
-# 
+# 	yaw 角速度惩罚（让它别自转）
+def yaw_rate_l2(env, asset_cfg=SceneEntityCfg("robot")):
+    asset = env.scene[asset_cfg.name]
+    return asset.data.root_ang_vel_w[:, 2] ** 2
+
+# 	侧向速度惩罚（让它别横着走）
+def lin_vel_y_l2(env, asset_cfg=SceneEntityCfg("robot")):
+    asset = env.scene[asset_cfg.name]
+    return asset.data.root_lin_vel_w[:, 1] ** 2
+
+
+def feet_clearance_reward(
+    env, asset_cfg: SceneEntityCfg,
+    sensor_cfg: SceneEntityCfg,
+    target_height: float = 0.05,
+    force_threshold: float = 5.0,
+):
+    robot = env.scene[asset_cfg.name]
+
+    # ===== DEBUG PRINT (only once) =====
+    if not hasattr(env, "_printed_foot_quat"):
+        foot_quat_w = robot.data.body_quat_w[:, asset_cfg.body_ids, :]  # [N,2,4]
+        print("DEBUG foot body_ids:", asset_cfg.body_ids)
+        print("DEBUG LEFT foot quat:", foot_quat_w[0, 0])
+        print("DEBUG RIGHT foot quat:", foot_quat_w[0, 1])
+        print("DEBUG ROOT quat:", robot.data.root_quat_w[0])
+        env._printed_foot_quat = True
+
+    feet_pos_w = robot.data.body_pos_w[:, asset_cfg.body_ids, :]
+    foot_z = feet_pos_w[..., 2]  # [num_envs, 2]
+
+    # infer contact from forces
+    sensor = env.scene.sensors[sensor_cfg.name]
+    forces_w = sensor.data.net_forces_w
+    force_mag = torch.linalg.norm(forces_w[:, sensor_cfg.body_ids, :], dim=-1)  # [num_envs, 2]
+    in_contact = force_mag > force_threshold
+
+    # Only reward clearance when foot is NOT in contact (swing foot)
+    swing = ~in_contact
+
+    # hinge reward up to target_height
+    z_clamped = torch.clamp(foot_z, 0.0, target_height) / target_height
+
+    # mask: stance foot gets 0 reward
+    reward = torch.sum(z_clamped * swing.float(), dim=-1)
+    return reward
+
+def knee_flexion_target_exp(
+    env,
+    asset_cfg: SceneEntityCfg,
+    knee_target: float = 0.8,   # 目标屈膝角度（你先从 0.6~1.0 试）
+    std: float = 0.3,
+):
+    asset: Articulation = env.scene[asset_cfg.name]
+    q = asset.data.joint_pos[:, asset_cfg.joint_ids]  # [N, num_selected]
+    err = q - knee_target
+    # 每个env对选中的膝关节做平均
+    return torch.exp(-torch.mean(err * err, dim=-1) / (std * std))
+
+def swing_knee_flexion_reward(
+    env,
+    sensor_cfg: SceneEntityCfg,
+    knee_cfg: SceneEntityCfg,
+    knee_target: float = 0.9,
+    std: float = 0.3,
+    force_threshold: float = 10.0,
+):
+    sensor: ContactSensor = env.scene.sensors[sensor_cfg.name]
+    forces = sensor.data.net_forces_w[:, sensor_cfg.body_ids, :]   # [N,2,3]
+    in_contact = torch.linalg.norm(forces, dim=-1) > force_threshold  # [N,2]
+    swing = ~in_contact  # [N,2]
+
+    asset: Articulation = env.scene[knee_cfg.name]
+    qk = asset.data.joint_pos[:, knee_cfg.joint_ids]  # [N,2] 选中左右膝
 
+    # 只在 swing 的那条腿上算奖励
+    err = qk - knee_target
+    r = torch.exp(-(err * err) / (std * std))  # [N,2]
+    r = torch.sum(r * swing.float(), dim=-1)   # [N]
+
+    return r
 
 # Unused reward terms (currently disabled in env cfg)
 # Keep here for future experiments.
@@ -379,42 +458,8 @@ def feet_lateral_separation_reward(
 #     return penalty
 
 
-# def feet_clearance_reward(
-#     env, asset_cfg: SceneEntityCfg,
-#     sensor_cfg: SceneEntityCfg,
-#     target_height: float = 0.05,
-#     force_threshold: float = 5.0,
-# ):
-#     robot = env.scene[asset_cfg.name]
-
-#     # ===== DEBUG PRINT (only once) =====
-#     if not hasattr(env, "_printed_foot_quat"):
-#         foot_quat_w = robot.data.body_quat_w[:, asset_cfg.body_ids, :]  # [N,2,4]
-#         print("DEBUG foot body_ids:", asset_cfg.body_ids)
-#         print("DEBUG LEFT foot quat:", foot_quat_w[0, 0])
-#         print("DEBUG RIGHT foot quat:", foot_quat_w[0, 1])
-#         print("DEBUG ROOT quat:", robot.data.root_quat_w[0])
-#         env._printed_foot_quat = True
-
-#     feet_pos_w = robot.data.body_pos_w[:, asset_cfg.body_ids, :]
-#     foot_z = feet_pos_w[..., 2]  # [num_envs, 2]
-
-#     # infer contact from forces
-#     sensor = env.scene.sensors[sensor_cfg.name]
-#     forces_w = sensor.data.net_forces_w
-#     force_mag = torch.linalg.norm(forces_w[:, sensor_cfg.body_ids, :], dim=-1)  # [num_envs, 2]
-#     in_contact = force_mag > force_threshold
-
-#     # Only reward clearance when foot is NOT in contact (swing foot)
-#     swing = ~in_contact
 
-#     # hinge reward up to target_height
-#     z_clamped = torch.clamp(foot_z, 0.0, target_height) / target_height
 
-#     # mask: stance foot gets 0 reward
-#     reward = torch.sum(z_clamped * swing.float(), dim=-1)
-
-#     return reward
 
 # class joint_pos_limits_penalty_ratio(ManagerTermBase):
 #     """Penalty for violating joint position limits weighted by the gear ratio."""