--- git status ---
On branch joystick-tracking
Your branch is up to date with 'origin/joystick-tracking'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	logs/rsl_rl/dodo_managed_ppo/2026-01-08_14-00-30/
	logs/rsl_rl/dodo_managed_ppo/2026-01-08_14-21-15/
	logs/rsl_rl/dodo_managed_ppo/2026-01-08_15-10-10/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
index 45132ae..f211fb8 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
@@ -63,7 +63,7 @@ class CommandsCfg:
     base_velocity = mdp.UniformVelocityCommandCfg(
         asset_name="robot",
         resampling_time_range=(10.0, 10.0),
-        rel_standing_envs=0.02, # 2%的环境将获得零速度命令（站立不动）
+        rel_standing_envs=0.0, # 没有环境使用站立控制
         rel_heading_envs=1.0, # 100%的环境将使用朝向控制
         heading_command=True, # 朝向命令
         heading_control_stiffness=0.5, # 刚度系数，响应朝向变化的速度
@@ -87,7 +87,7 @@ class ActionsCfg:
     joint_effort = mdp.JointEffortActionCfg(
         asset_name="robot",
         joint_names=[".*"],
-        scale=2.5
+        scale=1.0
     )
 
 
@@ -189,24 +189,26 @@ class EventCfg:
         },
     )
 
-
+# Rewards
 @configclass
 class RewardsCfg:
     """Reward terms for the MDP."""
 
-# isaaclab自带
-    alive = RewTerm(func=mdp.is_alive, weight=2.0) # 存活奖励
-    termination = RewTerm(func=mdp.is_terminated, weight=-10.0) # 结束惩罚
-    action_l2 = RewTerm(func=mdp.action_l2, weight=-0.01) # 惩罚过大的动作
+    #alive reward
+    alive = RewTerm(func=mdp.is_alive, weight=2.0)
+    #termination penalty
+    termination = RewTerm(func=mdp.is_terminated, weight=-10.0)
+    #action penalty
+    action_l2 = RewTerm(func=mdp.action_l2, weight=-0.01)
 
     # 髋关节移位置惩罚
-    hip_joint_move = RewTerm(func=mdp.hip_pos_manual_limit,
-        weight=-10.0,
-        params={
-            "asset_cfg": SceneEntityCfg("robot", joint_names=["left_joint_1", "right_joint_1"]),
-            "softlimit": (0, 0),
-        },
-    )
+    # hip_joint_move = RewTerm(func=mdp.hip_pos_manual_limit,
+    #     weight=-10.0,
+    #     params={
+    #         "asset_cfg": SceneEntityCfg("robot", joint_names=["left_joint_1", "right_joint_1"]),
+    #         "softlimit": (0, 0),
+    #     },
+    # )
 
 # 自己编写的
     # # 前进进度奖励
@@ -224,36 +226,37 @@ class RewardsCfg:
     track_ang_vel_z_exp = RewTerm(
         func=mdp.track_ang_vel_z_world_exp, weight=3.0, params={"command_name": "base_velocity", "std": 0.3}
     )
-    # 能耗惩罚
-    energy = RewTerm(
-        func=mdp.power_consumption, weight=-0.005,
-        params={"gear_ratio": {".*": 2.5}},
-    )
-    # 关节极限惩罚
-    joint_pos_limits = RewTerm(
-        func=mdp.joint_pos_limits_penalty_ratio, weight=-0.25,
-        params={"threshold": 0.98, "gear_ratio": {".*": 2.5}},
-    )
-
-    feet_air_time = RewTerm(
-        func=mdp.feet_air_time_positive_biped_snesor,
-        weight=2.5,
-        params={
-            "command_name": "base_velocity",
-            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
-            "threshold": 2.5,
-        },
-    )
+    # # 能耗惩罚
+    # energy = RewTerm(
+    #     func=mdp.power_consumption, weight=-0.005,
+    #     params={"gear_ratio": {".*": 2.5}},
+    # )
+    # # 关节极限惩罚
+    # joint_pos_limits = RewTerm(
+    #     func=mdp.joint_pos_limits_penalty_ratio, weight=-0.25,
+    #     params={"threshold": 0.98, "gear_ratio": {".*": 2.5}},
+    # )
+
+    # feet_air_time = RewTerm(
+    #     func=mdp.feet_air_time_positive_biped_snesor,
+    #     weight=2.5,
+    #     params={
+    #         "command_name": "base_velocity",
+    #         "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
+    #         "threshold": 2.5,
+    #     },
+    # )
         
-    feet_slide = RewTerm(
-        func=mdp.feet_slide,
-        weight=-0.5,
-        params={
-            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
-            "asset_cfg": SceneEntityCfg("robot", body_names=".*_link_4"),
-        },
-    )
-
+    # feet_slide = RewTerm(
+    #     func=mdp.feet_slide,
+    #     weight=-0.5,
+    #     params={
+    #         "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
+    #         "asset_cfg": SceneEntityCfg("robot", body_names=".*_link_4"),
+    #     },
+    # )
+
+# Terminations
 @configclass
 class TerminationsCfg:
     """Termination terms for the MDP."""
@@ -271,10 +274,10 @@ class TerminationsCfg:
         params={"asset_cfg": SceneEntityCfg("robot"), "limit_angle": 0.8},
     )
 
-    hip_threshold = DoneTerm(
-        func=mdp.joint_pos_out_of_manual_limit,
-        params={"asset_cfg": SceneEntityCfg("robot", joint_names=["left_joint_1", "right_joint_1"]), "bounds": [-0.15, 0.15]},
-    )
+    # hip_threshold = DoneTerm(
+    #     func=mdp.joint_pos_out_of_manual_limit,
+    #     params={"asset_cfg": SceneEntityCfg("robot", joint_names=["left_joint_1", "right_joint_1"]), "bounds": [-0.15, 0.15]},
+    # )
 
 
 @configclass