--- git status ---
On branch joystick-tracking
Your branch is up to date with 'origin/joystick-tracking'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	logs/rsl_rl/dodo_managed_ppo/2026-01-08_14-00-30/
	logs/rsl_rl/dodo_managed_ppo/2026-01-08_14-21-15/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
index 45132ae..517e917 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
@@ -200,13 +200,13 @@ class RewardsCfg:
     action_l2 = RewTerm(func=mdp.action_l2, weight=-0.01) # 惩罚过大的动作
 
     # 髋关节移位置惩罚
-    hip_joint_move = RewTerm(func=mdp.hip_pos_manual_limit,
-        weight=-10.0,
-        params={
-            "asset_cfg": SceneEntityCfg("robot", joint_names=["left_joint_1", "right_joint_1"]),
-            "softlimit": (0, 0),
-        },
-    )
+    # hip_joint_move = RewTerm(func=mdp.hip_pos_manual_limit,
+    #     weight=-10.0,
+    #     params={
+    #         "asset_cfg": SceneEntityCfg("robot", joint_names=["left_joint_1", "right_joint_1"]),
+    #         "softlimit": (0, 0),
+    #     },
+    # )
 
 # 自己编写的
     # # 前进进度奖励
@@ -235,15 +235,15 @@ class RewardsCfg:
         params={"threshold": 0.98, "gear_ratio": {".*": 2.5}},
     )
 
-    feet_air_time = RewTerm(
-        func=mdp.feet_air_time_positive_biped_snesor,
-        weight=2.5,
-        params={
-            "command_name": "base_velocity",
-            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
-            "threshold": 2.5,
-        },
-    )
+    # feet_air_time = RewTerm(
+    #     func=mdp.feet_air_time_positive_biped_snesor,
+    #     weight=2.5,
+    #     params={
+    #         "command_name": "base_velocity",
+    #         "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
+    #         "threshold": 2.5,
+    #     },
+    # )
         
     feet_slide = RewTerm(
         func=mdp.feet_slide,
@@ -271,10 +271,10 @@ class TerminationsCfg:
         params={"asset_cfg": SceneEntityCfg("robot"), "limit_angle": 0.8},
     )
 
-    hip_threshold = DoneTerm(
-        func=mdp.joint_pos_out_of_manual_limit,
-        params={"asset_cfg": SceneEntityCfg("robot", joint_names=["left_joint_1", "right_joint_1"]), "bounds": [-0.15, 0.15]},
-    )
+    # hip_threshold = DoneTerm(
+    #     func=mdp.joint_pos_out_of_manual_limit,
+    #     params={"asset_cfg": SceneEntityCfg("robot", joint_names=["left_joint_1", "right_joint_1"]), "bounds": [-0.15, 0.15]},
+    # )
 
 
 @configclass