--- git status ---
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/events.out.tfevents.1763052126.ray-ubuntu.30173.0
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/exported/policy.onnx
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/exported/policy.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/git/isaac_dodo.diff
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_0.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_100.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_150.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_200.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_250.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_300.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_350.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_400.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_450.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_50.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_500.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_550.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_600.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_650.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_700.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_750.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_800.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_850.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_900.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_950.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_999.pt
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/agent.pkl
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/agent.yaml
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/env.pkl
	deleted:    logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/env.yaml
	modified:   source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py
	modified:   source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py
	modified:   source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
	modified:   source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_19-00-19/exported/
	logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_16-56-29/
	logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_17-13-33/
	logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_17-51-34/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/events.out.tfevents.1763052126.ray-ubuntu.30173.0 b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/events.out.tfevents.1763052126.ray-ubuntu.30173.0
deleted file mode 100644
index ed1b781..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/events.out.tfevents.1763052126.ray-ubuntu.30173.0 and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/exported/policy.onnx b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/exported/policy.onnx
deleted file mode 100644
index f604c9d..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/exported/policy.onnx and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/exported/policy.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/exported/policy.pt
deleted file mode 100644
index 5a3f7c1..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/exported/policy.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/git/isaac_dodo.diff b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/git/isaac_dodo.diff
deleted file mode 100644
index 0525043..0000000
--- a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/git/isaac_dodo.diff
+++ /dev/null
@@ -1,469 +0,0 @@
---- git status ---
-On branch main
-Your branch is up to date with 'origin/main'.
-
-Changes not staged for commit:
-  (use "git add <file>..." to update what will be committed)
-  (use "git restore <file>..." to discard changes in working directory)
-	modified:   source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py
-	modified:   source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py
-
-Untracked files:
-  (use "git add <file>..." to include in what will be committed)
-	source/isaac_dodo/isaac_dodo/assets/
-
-no changes added to commit (use "git add" and/or "git commit -a") 
-
-
---- git diff ---
-diff --git a/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py b/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py
-index 5ce9cc5..981a4ba 100644
---- a/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py
-+++ b/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py
-@@ -32,18 +32,13 @@ class DodoVelocityTrackingEnv(LocomotionEnv):
-         self.motor_effort_ratio = torch.ones_like(self.joint_gears, device=self.sim.device)
-         self._joint_dof_idx, _ = self.robot.find_joints(".*")
- 
--        # remove positional target; use pure speed+heading targets (per-env)
--        self.target_forward_speeds = torch.full((self.num_envs,), float(self.cfg.target_forward_speed), device=self.sim.device)
--        # desired heading (radians, 0 = +x); keep fixed 0 unless randomized elsewhere
--        self.target_headings = torch.zeros((self.num_envs,), dtype=torch.float32, device=self.sim.device)
--        self.start_rotation = torch.tensor([1, 0, 0, 0], device=self.sim.device, dtype=torch.float32)
--        self.up_vec = torch.tensor([0, 0, 1], dtype=torch.float32, device=self.sim.device).repeat((self.num_envs, 1))
--        self.heading_vec = torch.tensor([1, 0, 0], dtype=torch.float32, device=self.sim.device).repeat(
--            (self.num_envs, 1)
--        )
--        self.inv_start_rot = quat_conjugate(self.start_rotation).repeat((self.num_envs, 1))
--        self.basis_vec0 = self.heading_vec.clone()
--        self.basis_vec1 = self.up_vec.clone()
-+        lin_low, lin_high = self.cfg.cmd_lin_range
-+        ang_low, ang_high = self.cfg.cmd_ang_range
-+        self.command_lin = lin_low + (lin_high - lin_low) * torch.rand((self.num_envs, 1), device=self.sim.device)
-+        self.command_ang = ang_low + (ang_high - ang_low) * torch.rand((self.num_envs, 1), device=self.sim.device)
-+
-+        self.applied_torques = torch.zeros((self.num_envs, self.cfg.action_space), dtype=torch.float32, device=self.sim.device)
-+        self.prev_actions = torch.zeros((self.num_envs, self.cfg.action_space), dtype=torch.float32, device=self.sim.device)
- 
-     def _setup_scene(self):
-         self.robot = Articulation(self.cfg.robot)
-@@ -63,64 +58,42 @@ class DodoVelocityTrackingEnv(LocomotionEnv):
-         light_cfg.func("/World/Light", light_cfg)
- 
-     def _pre_physics_step(self, actions: torch.Tensor):
-+        self.prev_actions[:] = getattr(self, "actions", torch.zeros_like(actions))
-         self.actions = actions.clone()
- 
-     def _apply_action(self):
-         forces = self.action_scale * self.joint_gears * self.actions
-+        self.applied_torques = forces.clone()
-         self.robot.set_joint_effort_target(forces, joint_ids=self._joint_dof_idx)
- 
-     def _compute_intermediate_values(self):
--        self.torso_position, self.torso_rotation = self.robot.data.root_pos_w, self.robot.data.root_quat_w
--        self.velocity, self.ang_velocity = self.robot.data.root_lin_vel_w, self.robot.data.root_ang_vel_w
--        self.dof_pos, self.dof_vel = self.robot.data.joint_pos, self.robot.data.joint_vel
--
--        (
--            self.up_proj,
--            self.heading_proj,
--            self.up_vec,
--            self.heading_vec,
--            self.vel_loc,
--            self.angvel_loc,
--            self.roll,
--            self.pitch,
--            self.yaw,
--            self.angle_to_target,
--            self.dof_pos_scaled,
--            self.prev_potentials,
--            self.potentials,
--        ) = compute_intermediate_values(
--            self.targets,
--            self.torso_position,
--            self.torso_rotation,
--            self.velocity,
--            self.ang_velocity,
--            self.dof_pos,
--            self.robot.data.soft_joint_pos_limits[0, :, 0],
--            self.robot.data.soft_joint_pos_limits[0, :, 1],
--            self.inv_start_rot,
--            self.basis_vec0,
--            self.basis_vec1,
--            self.potentials,
--            self.prev_potentials,
--            self.cfg.sim.dt,
--        )
-+        self.motor_pos = self.robot.data.joint_pos          # 8 joint positions
-+        self.motor_vel = self.robot.data.joint_vel          # 8 joint velocities
-+        self.motor_tau = self.applied_torques               # 8 joint efforts
-+
-+        self.base_lin_vel = self.robot.data.root_lin_vel_w  # 3 base linear velocities
-+        self.base_ang_vel = self.robot.data.root_ang_vel_w  # 3 base angular velocities
-+
-+        self.base_position = self.robot.data.root_pos_w     # base position
-+        q = self.robot.data.root_quat_w                     # roll, pitch, yaw
-+        qw, qx, qy, qz = q[:, 0], q[:, 1], q[:, 2], q[:, 3]
-+        self.roll = torch.atan2(2 * (qw * qx + qy * qz), 1 - 2 * (qx * qx + qy * qy))
-+        self.pitch = torch.asin(torch.clamp(2 * (qw * qy - qz * qx), -1 + 1e-6, 1 - 1e-6))
-+        self.yaw = torch.atan2(2 * (qw * qz + qx * qy), 1 - 2 * (qy * qy + qz * qz))
- 
-     def _get_observations(self) -> dict:
-         obs = torch.cat(
-             (
--                self.torso_position[:, 2].view(-1, 1),
--                self.vel_loc,
--                self.ang_velocity * self.cfg.angular_velocity_scale,
--                normalize_angle(self.yaw).unsqueeze(-1),
--                normalize_angle(self.roll).unsqueeze(-1),
--                normalize_angle(self.angle_to_target).unsqueeze(-1),
--                self.up_proj.unsqueeze(-1),
--                self.heading_proj.unsqueeze(-1),
--                self.dof_pos_scaled,
--                self.dof_vel * self.cfg.dof_vel_scale,
--                self.actions,
--                self.target_forward_speeds.unsqueeze(-1),
--                self.target_headings.unsqueeze(-1),
-+                self.motor_pos,
-+                self.motor_vel,
-+                self.motor_tau,
-+                self.base_lin_vel,
-+                self.base_ang_vel,
-+                self.roll.unsqueeze(-1),
-+                self.pitch.unsqueeze(-1),
-+                self.yaw.unsqueeze(-1),
-+                self.command_lin,
-+                self.command_ang,
-             ),
-             dim=-1,
-         )
-@@ -128,172 +101,76 @@ class DodoVelocityTrackingEnv(LocomotionEnv):
-         return observations
- 
-     def _get_rewards(self) -> torch.Tensor:
--        total_reward = compute_rewards(
--            self.actions,
--            self.reset_terminated,
--            self.cfg.up_weight,
--            self.cfg.heading_weight,
--            self.heading_proj,
--            self.up_proj,
--            self.dof_vel,
--            self.dof_pos_scaled,
--            self.cfg.actions_cost_scale,
--            self.cfg.energy_cost_scale,
--            self.cfg.dof_vel_scale,
--            self.cfg.death_cost,
--            self.cfg.alive_reward_scale,
--            self.motor_effort_ratio,
--            self.vel_loc,
--            self.angle_to_target,
--            self.target_forward_speeds,
--            float(self.cfg.vel_tracking_weight),
--            float(self.cfg.dir_tracking_weight),
--        )
-+        self._compute_intermediate_values()
-+
-+        # linear and angular velocity tracking
-+        v_forward = torch.sqrt(self.base_lin_vel[:, 0] ** 2 + self.base_lin_vel[:, 1] ** 2)
-+        v_command = self.command_lin.squeeze(-1)
-+        w_yaw = self.base_ang_vel[:, 2]
-+        w_command = self.command_ang.squeeze(-1)
-+
-+        lin_vel_err_sq = (v_forward - v_command) ** 2
-+        r_lin = self.cfg.reward_lin_vel_w * torch.exp(-lin_vel_err_sq / (2 * self.cfg.lin_vel_sigma ** 2))
-+        ang_vel_err_sq = (w_yaw - w_command) ** 2
-+        r_ang = self.cfg.reward_ang_vel_w * torch.exp(-ang_vel_err_sq / (2 * self.cfg.ang_vel_sigma ** 2))
-+
-+        # orientation reward
-+        orient_err_sq = self.roll ** 2 + self.pitch ** 2
-+        r_orient = self.cfg.reward_orientation_w * torch.exp(-orient_err_sq / (2 * self.cfg.orientation_sigma ** 2))
-+
-+        # energy regularization
-+        torque_mag = torch.sum(torch.abs(self.motor_tau), dim=1)           # L1
-+        max_possible = torch.sum(self.joint_gears) * self.action_scale
-+        r_torque = self.cfg.reward_torque_reg_w * (1.0 - torque_mag / (max_possible + 1e-6)).clamp(min=0.0)
-+
-+        # action rate regularization
-+        action_diff_sq = torch.mean((self.actions - self.prev_actions) ** 2, dim=1)
-+        r_action_rate = self.cfg.reward_action_rate_w * torch.exp(-action_diff_sq / (2 * self.cfg.action_rate_sigma ** 2))
-+
-+        # alive 
-+        r_alive = torch.ones(self.num_envs, device=self.sim.device) * self.cfg.reward_alive_w
-+
-+        # failure penalty
-+        height_failure = self.base_position[:, 2] < self.cfg.termination_height
-+        roll_failure = torch.abs(self.roll) > self.cfg.termination_roll
-+        pitch_failure = torch.abs(self.pitch) > self.cfg.termination_pitch
-+        failure = height_failure | roll_failure | pitch_failure
-+        p_fail = failure.float() * self.cfg.reward_failure_penalty
-+
-+        # total reward
-+        total_reward = r_lin + r_ang + r_orient + r_torque + r_action_rate + r_alive + p_fail
-+
-         return total_reward
- 
-     def _get_dones(self) -> tuple[torch.Tensor, torch.Tensor]:
-         self._compute_intermediate_values()
-         time_out = self.episode_length_buf >= self.max_episode_length - 1
--        died = self.torso_position[:, 2] < self.cfg.termination_height
--        return time_out, died
-+        height_failure = self.base_position[:, 2] < self.cfg.termination_height
-+        roll_failure = torch.abs(self.roll) > self.cfg.termination_roll
-+        pitch_failure = torch.abs(self.pitch) > self.cfg.termination_pitch
-+        died = height_failure | roll_failure | pitch_failure
-+        return died, time_out
- 
-     def _reset_idx(self, env_ids: torch.Tensor | None):
--        # ensure env_ids is a valid index tensor for per-env operations
--        if env_ids is None:
--            env_ids = torch.arange(self.num_envs, device=self.sim.device, dtype=torch.long)
-+        if env_ids is None or len(env_ids) == self.num_envs:
-+            env_ids = self.robot._ALL_INDICES
-         self.robot.reset(env_ids)
-         super()._reset_idx(env_ids)
- 
-+        # reset robot state
-         joint_pos = self.robot.data.default_joint_pos[env_ids]
-         joint_vel = self.robot.data.default_joint_vel[env_ids]
--        default_root_state = self.robot.data.default_root_state[env_ids]
--        default_root_state[:, :3] += self.scene.env_origins[env_ids]
--
--        self.robot.write_root_pose_to_sim(default_root_state[:, :7], env_ids)
--        self.robot.write_root_velocity_to_sim(default_root_state[:, 7:], env_ids)
-+        root_state = self.robot.data.default_root_state[env_ids]
-+        root_state[:, :3] += self.scene.env_origins[env_ids]
-+        self.robot.write_root_pose_to_sim(root_state[:, :7], env_ids)
-+        self.robot.write_root_velocity_to_sim(root_state[:, 7:], env_ids)
-         self.robot.write_joint_state_to_sim(joint_pos, joint_vel, None, env_ids)
- 
--        # ensure per-env targets exist for reset envs
--        # initial randomization: speed in [-1, 2] m/s, heading in [0, 2*pi)
--        self.target_forward_speeds[env_ids] = torch.empty(
--            len(env_ids), device=self.sim.device
--        ).uniform_(-2.0, 2.0)
--        self.target_headings[env_ids] = torch.empty(
--            len(env_ids), device=self.sim.device
--        ).uniform_(0.0, 2.0 * math.pi)
-+        # new command
-+        lin_low, lin_high = self.cfg.cmd_lin_range
-+        ang_low, ang_high = self.cfg.cmd_ang_range
-+        self.command_lin[env_ids] = lin_low + (lin_high - lin_low) * torch.rand((len(env_ids), 1), device=self.sim.device)
-+        self.command_ang[env_ids] = ang_low + (ang_high - ang_low) * torch.rand((len(env_ids), 1), device=self.sim.device)
- 
-+        self.applied_torques[env_ids] = 0.0
-         self._compute_intermediate_values()
--
--
--@torch.jit.script
--def compute_rewards(
--    actions: torch.Tensor,
--    reset_terminated: torch.Tensor,
--    up_weight: float,
--    heading_weight: float,
--    heading_proj: torch.Tensor,
--    up_proj: torch.Tensor,
--    dof_vel: torch.Tensor,
--    dof_pos_scaled: torch.Tensor,
--    actions_cost_scale: float,
--    energy_cost_scale: float,
--    dof_vel_scale: float,
--    death_cost: float,
--    alive_reward_scale: float,
--    motor_effort_ratio: torch.Tensor,
--    vel_loc: torch.Tensor,
--    angle_to_target: torch.Tensor,
--    target_forward_speed: torch.Tensor,
--    vel_tracking_weight: float,
--    dir_tracking_weight: float,
--):
--    heading_weight_tensor = torch.ones_like(heading_proj) * heading_weight
--    heading_reward = torch.where(heading_proj > 0.8, heading_weight_tensor, heading_weight * heading_proj / 0.8)
--
--    # aligning up axis of robot and environment
--    up_reward = torch.zeros_like(heading_reward)
--    up_reward = torch.where(up_proj > 0.95, up_reward + up_weight, up_reward)
--
--    # energy / action penalties
--    actions_cost = torch.sum(actions**2, dim=-1)
--    electricity_cost = torch.sum(
--        torch.abs(actions * dof_vel * dof_vel_scale) * motor_effort_ratio.unsqueeze(0),
--        dim=-1,
--    )
--    dof_at_limit_cost = torch.sum(dof_pos_scaled > 0.98, dim=-1)
--
--    # alive reward (per step)
--    alive_reward = torch.ones_like(heading_proj) * alive_reward_scale
--
--    # velocity tracking (forward axis = x in local frame)
--    vel_forward = vel_loc[:, 0]
--    vel_err = vel_forward - target_forward_speed
--    vel_tracking_reward = -vel_tracking_weight * (vel_err * vel_err)
--
--    # direction tracking: cosine of heading error
--    dir_tracking_reward = dir_tracking_weight * torch.cos(angle_to_target)
--
--    total_reward = (
--        vel_tracking_reward
--        + dir_tracking_reward
--        + alive_reward
--        + up_reward
--        + heading_reward
--        - actions_cost_scale * actions_cost
--        - energy_cost_scale * electricity_cost
--        - dof_at_limit_cost
--    )
--    total_reward = torch.where(reset_terminated, torch.ones_like(total_reward) * death_cost, total_reward)
--    return total_reward
--
--
--@torch.jit.script
--def compute_intermediate_values(
--    targets: torch.Tensor,
--    torso_position: torch.Tensor,
--    torso_rotation: torch.Tensor,
--    velocity: torch.Tensor,
--    ang_velocity: torch.Tensor,
--    dof_pos: torch.Tensor,
--    dof_lower_limits: torch.Tensor,
--    dof_upper_limits: torch.Tensor,
--    inv_start_rot: torch.Tensor,
--    basis_vec0: torch.Tensor,
--    basis_vec1: torch.Tensor,
--    potentials: torch.Tensor,
--    prev_potentials: torch.Tensor,
--    dt: float,
--):
--    to_target = targets - torso_position
--    to_target[:, 2] = 0.0
--
--    torso_quat, up_proj, heading_proj, up_vec, heading_vec = compute_heading_and_up(
--        torso_rotation, inv_start_rot, to_target, basis_vec0, basis_vec1, 2
--    )
--
--    vel_loc, angvel_loc, roll, pitch, yaw, angle_to_target = compute_rot(
--        torso_quat, velocity, ang_velocity, targets, torso_position
--    )
--
--    dof_pos_scaled = torch_utils.maths.unscale(dof_pos, dof_lower_limits, dof_upper_limits)
--
--    to_target = targets - torso_position
--    to_target[:, 2] = 0.0
--    prev_potentials[:] = potentials
--    potentials = -torch.norm(to_target, p=2, dim=-1) / dt
--
--    return (
--        up_proj,
--        heading_proj,
--        up_vec,
--        heading_vec,
--        vel_loc,
--        angvel_loc,
--        roll,
--        pitch,
--        yaw,
--        angle_to_target,
--        dof_pos_scaled,
--        prev_potentials,
--        potentials,
--    )
-diff --git a/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py b/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py
-index 0f909fb..e4e3613 100644
---- a/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py
-+++ b/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py
-@@ -3,7 +3,7 @@
- #
- # SPDX-License-Identifier: BSD-3-Clause
- 
--from isaaclab_assets import HUMANOID_CFG
-+from isaac_dodo.assets.robots.dodo import DODO_CFG
- 
- import isaaclab.sim as sim_utils
- from isaaclab.assets import ArticulationCfg
-@@ -20,9 +20,17 @@ class DodoVelocityTrackingEnvCfg(DirectRLEnvCfg):
-     episode_length_s = 15.0
-     decimation = 2
-     action_scale = 1.0
--    action_space = 21
--    observation_space = 77
--    state_space = 2
-+    action_space = 8
-+    observation_space = 35
-+    # 8 joint positions
-+    # 8 joint velocities
-+    # 8 joint efforts
-+    # 3 base linear velocities
-+    # 3 base angular velocities
-+    # roll, pitch, yaw
-+    # linear velocity command (1 dim)
-+    # angular velocity command (1 dim)
-+    state_space = 0
- 
-     # simulation
-     sim: SimulationCfg = SimulationCfg(dt=1 / 120, render_interval=decimation)
-@@ -46,44 +54,30 @@ class DodoVelocityTrackingEnvCfg(DirectRLEnvCfg):
-     )
- 
-     # robot
--    robot: ArticulationCfg = HUMANOID_CFG.replace(prim_path="/World/envs/env_.*/Robot")
-+    robot: ArticulationCfg = DODO_CFG.replace(prim_path="/World/envs/env_.*/Robot")
-     joint_gears: list = [
--        67.5000,  # lower_waist
--        67.5000,  # lower_waist
--        67.5000,  # right_upper_arm
--        67.5000,  # right_upper_arm
--        67.5000,  # left_upper_arm
--        67.5000,  # left_upper_arm
--        67.5000,  # pelvis
--        45.0000,  # right_lower_arm
--        45.0000,  # left_lower_arm
--        45.0000,  # right_thigh: x
--        135.0000,  # right_thigh: y
--        45.0000,  # right_thigh: z
--        45.0000,  # left_thigh: x
--        135.0000,  # left_thigh: y
--        45.0000,  # left_thigh: z
--        90.0000,  # right_knee
--        90.0000,  # left_knee
--        22.5,  # right_foot
--        22.5,  # right_foot
--        22.5,  # left_foot
--        22.5,  # left_foot
-+        2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5, 2.5
-     ]
--    target_forward_speed: float = 1.0
--    vel_tracking_weight: float = 3.0
--    dir_tracking_weight: float = 2.5
- 
--    heading_weight: float = 0.5
--    up_weight: float = 0.1
-+    # task
-+    cmd_lin_range = [-1.0, 1.0]  # m/s
-+    cmd_ang_range = [-1.0, 1.0]  # rad/s
- 
--    energy_cost_scale: float = 0.05
--    actions_cost_scale: float = 0.01
--    alive_reward_scale: float = 2.0
--    dof_vel_scale: float = 0.1
-+    termination_height = 0.3  # m
-+    termination_roll = 1.0  # rad
-+    termination_pitch = 1.0  # rad
- 
--    death_cost: float = -1.0
--    termination_height: float = 1.1
-+    # reward scales
-+    reward_lin_vel_w = 2.0
-+    reward_ang_vel_w = 2.0
-+    reward_orientation_w = 0.5
-+    reward_torque_reg_w = 0.01
-+    reward_action_rate_w = 0.05
-+    reward_alive_w = 0.2
-+    reward_failure_penalty = -100.0
- 
--    angular_velocity_scale: float = 0.25
--    contact_force_scale: float = 0.01
-\ No newline at end of file
-+    # shaping sigmas
-+    lin_vel_sigma = 0.5
-+    ang_vel_sigma = 0.5
-+    orientation_sigma = 0.5
-+    action_rate_sigma = 0.2
\ No newline at end of file
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_0.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_0.pt
deleted file mode 100644
index a9a030e..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_0.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_100.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_100.pt
deleted file mode 100644
index 26c4d37..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_100.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_150.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_150.pt
deleted file mode 100644
index 6f1f7a5..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_150.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_200.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_200.pt
deleted file mode 100644
index 52c9532..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_200.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_250.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_250.pt
deleted file mode 100644
index 19ea52c..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_250.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_300.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_300.pt
deleted file mode 100644
index dc32bfe..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_300.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_350.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_350.pt
deleted file mode 100644
index 7f4c6c3..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_350.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_400.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_400.pt
deleted file mode 100644
index 934818b..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_400.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_450.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_450.pt
deleted file mode 100644
index 7862566..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_450.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_50.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_50.pt
deleted file mode 100644
index cfca3f3..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_50.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_500.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_500.pt
deleted file mode 100644
index fd200e6..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_500.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_550.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_550.pt
deleted file mode 100644
index a192f9a..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_550.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_600.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_600.pt
deleted file mode 100644
index ad82b4c..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_600.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_650.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_650.pt
deleted file mode 100644
index 7d8a2ca..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_650.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_700.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_700.pt
deleted file mode 100644
index 2679e30..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_700.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_750.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_750.pt
deleted file mode 100644
index 6b8be32..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_750.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_800.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_800.pt
deleted file mode 100644
index 6dcecbe..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_800.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_850.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_850.pt
deleted file mode 100644
index e71d401..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_850.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_900.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_900.pt
deleted file mode 100644
index 2cad6a4..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_900.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_950.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_950.pt
deleted file mode 100644
index 9046352..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_950.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_999.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_999.pt
deleted file mode 100644
index d8b0bc8..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/model_999.pt and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/agent.pkl b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/agent.pkl
deleted file mode 100644
index 38b93dc..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/agent.pkl and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/agent.yaml b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/agent.yaml
deleted file mode 100644
index 27bedef..0000000
--- a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/agent.yaml
+++ /dev/null
@@ -1,45 +0,0 @@
-seed: 42
-device: cuda:0
-num_steps_per_env: 32
-max_iterations: 1000
-empirical_normalization: true
-policy:
-  class_name: ActorCritic
-  init_noise_std: 1.0
-  noise_std_type: scalar
-  actor_hidden_dims:
-  - 400
-  - 200
-  - 100
-  critic_hidden_dims:
-  - 400
-  - 200
-  - 100
-  activation: elu
-algorithm:
-  class_name: PPO
-  num_learning_epochs: 5
-  num_mini_batches: 4
-  learning_rate: 0.0001
-  schedule: adaptive
-  gamma: 0.99
-  lam: 0.95
-  entropy_coef: 0.0
-  desired_kl: 0.008
-  max_grad_norm: 1.0
-  value_loss_coef: 1.0
-  use_clipped_value_loss: true
-  clip_param: 0.2
-  normalize_advantage_per_mini_batch: false
-  symmetry_cfg: null
-  rnd_cfg: null
-clip_actions: null
-save_interval: 50
-experiment_name: dodo_velocity_tracking_ppo
-run_name: ''
-logger: tensorboard
-neptune_project: isaaclab
-wandb_project: isaaclab
-resume: false
-load_run: .*
-load_checkpoint: model_.*.pt
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/env.pkl b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/env.pkl
deleted file mode 100644
index 34ae084..0000000
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/env.pkl and /dev/null differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/env.yaml b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/env.yaml
deleted file mode 100644
index 2c6d10d..0000000
--- a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-13_17-41-59/params/env.yaml
+++ /dev/null
@@ -1,266 +0,0 @@
-viewer:
-  eye: !!python/tuple
-  - 7.5
-  - 7.5
-  - 7.5
-  lookat: !!python/tuple
-  - 0.0
-  - 0.0
-  - 0.0
-  cam_prim_path: /OmniverseKit_Persp
-  resolution: !!python/tuple
-  - 1280
-  - 720
-  origin_type: world
-  env_index: 0
-  asset_name: null
-  body_name: null
-sim:
-  physics_prim_path: /physicsScene
-  device: cuda:0
-  dt: 0.008333333333333333
-  render_interval: 2
-  gravity: !!python/tuple
-  - 0.0
-  - 0.0
-  - -9.81
-  enable_scene_query_support: false
-  use_fabric: true
-  physx:
-    solver_type: 1
-    min_position_iteration_count: 1
-    max_position_iteration_count: 255
-    min_velocity_iteration_count: 0
-    max_velocity_iteration_count: 255
-    enable_ccd: false
-    enable_stabilization: false
-    enable_enhanced_determinism: false
-    bounce_threshold_velocity: 0.5
-    friction_offset_threshold: 0.04
-    friction_correlation_distance: 0.025
-    gpu_max_rigid_contact_count: 8388608
-    gpu_max_rigid_patch_count: 163840
-    gpu_found_lost_pairs_capacity: 2097152
-    gpu_found_lost_aggregate_pairs_capacity: 33554432
-    gpu_total_aggregate_pairs_capacity: 2097152
-    gpu_collision_stack_size: 67108864
-    gpu_heap_capacity: 67108864
-    gpu_temp_buffer_capacity: 16777216
-    gpu_max_num_partitions: 8
-    gpu_max_soft_body_contacts: 1048576
-    gpu_max_particle_contacts: 1048576
-  physics_material:
-    func: isaaclab.sim.spawners.materials.physics_materials:spawn_rigid_body_material
-    static_friction: 0.5
-    dynamic_friction: 0.5
-    restitution: 0.0
-    friction_combine_mode: average
-    restitution_combine_mode: average
-    compliant_contact_stiffness: 0.0
-    compliant_contact_damping: 0.0
-  render:
-    enable_translucency: null
-    enable_reflections: null
-    enable_global_illumination: null
-    antialiasing_mode: null
-    enable_dlssg: null
-    enable_dl_denoiser: null
-    dlss_mode: null
-    enable_direct_lighting: null
-    samples_per_pixel: null
-    enable_shadows: null
-    enable_ambient_occlusion: null
-    carb_settings: null
-    rendering_mode: null
-  create_stage_in_memory: false
-ui_window_class_type: isaaclab.envs.ui.base_env_window:BaseEnvWindow
-seed: 42
-decimation: 2
-is_finite_horizon: false
-episode_length_s: 15.0
-scene:
-  num_envs: 1024
-  env_spacing: 4.0
-  lazy_sensor_update: true
-  replicate_physics: true
-  filter_collisions: true
-  clone_in_fabric: true
-events: null
-observation_space: 35
-num_observations: null
-state_space: 0
-num_states: null
-observation_noise_model: null
-action_space: 8
-num_actions: null
-action_noise_model: null
-rerender_on_reset: false
-wait_for_textures: true
-xr: null
-robot:
-  class_type: isaaclab.assets.articulation.articulation:Articulation
-  prim_path: /World/envs/env_.*/Robot
-  spawn:
-    func: isaaclab.sim.spawners.from_files.from_files:spawn_from_usd
-    visible: true
-    semantic_tags: null
-    copy_from_source: true
-    mass_props: null
-    deformable_props: null
-    rigid_props:
-      rigid_body_enabled: true
-      kinematic_enabled: null
-      disable_gravity: false
-      linear_damping: 0.0
-      angular_damping: 0.0
-      max_linear_velocity: 1000.0
-      max_angular_velocity: 1000.0
-      max_depenetration_velocity: 1.0
-      max_contact_impulse: null
-      enable_gyroscopic_forces: null
-      retain_accelerations: false
-      solver_position_iteration_count: null
-      solver_velocity_iteration_count: null
-      sleep_threshold: null
-      stabilization_threshold: null
-    collision_props: null
-    activate_contact_sensors: false
-    scale: null
-    articulation_props:
-      articulation_enabled: null
-      enabled_self_collisions: false
-      solver_position_iteration_count: 8
-      solver_velocity_iteration_count: 4
-      sleep_threshold: null
-      stabilization_threshold: null
-      fix_root_link: null
-    fixed_tendons_props: null
-    spatial_tendons_props: null
-    joint_drive_props: null
-    visual_material_path: material
-    visual_material: null
-    usd_path: /home/ray/rl/isaac_dodo/source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd
-    variants: null
-  init_state:
-    pos: !!python/tuple
-    - 0.0
-    - 0.0
-    - 0.6
-    rot: !!python/tuple
-    - 1.0
-    - 0.0
-    - 0.0
-    - 0.0
-    lin_vel: !!python/tuple
-    - 0.0
-    - 0.0
-    - 0.0
-    ang_vel: !!python/tuple
-    - 0.0
-    - 0.0
-    - 0.0
-    joint_pos:
-      .*_joint_1: 0.0
-      .*_joint_2: -0.35
-      .*_joint_3: 0.7
-      .*_joint_4: -0.35
-    joint_vel:
-      .*: 0.0
-  collision_group: 0
-  debug_vis: false
-  articulation_root_prim_path: null
-  soft_joint_pos_limit_factor: 0.9
-  actuators:
-    legs:
-      class_type: isaaclab.actuators.actuator_pd:ImplicitActuator
-      joint_names_expr:
-      - .*_joint_1
-      - .*_joint_2
-      - .*_joint_3
-      - .*_joint_4
-      effort_limit: null
-      velocity_limit: null
-      effort_limit_sim: 2.5
-      velocity_limit_sim: null
-      stiffness:
-        .*_joint_1: 150.0
-        .*_joint_2: 150.0
-        .*_joint_3: 200.0
-        .*_joint_4: 200.0
-      damping:
-        .*_joint_1: 5.0
-        .*_joint_2: 5.0
-        .*_joint_3: 5.0
-        .*_joint_4: 5.0
-      armature:
-        .*_joint_1: 0.01
-        .*_joint_2: 0.01
-        .*_joint_3: 0.01
-        .*_joint_4: 0.01
-      friction: null
-      dynamic_friction: null
-      viscous_friction: null
-  actuator_value_resolution_debug_print: false
-joint_gears:
-- 2.5
-- 2.5
-- 2.5
-- 2.5
-- 2.5
-- 2.5
-- 2.5
-- 2.5
-action_scale: 1.0
-terrain:
-  class_type: isaaclab.terrains.terrain_importer:TerrainImporter
-  collision_group: -1
-  prim_path: /World/ground
-  num_envs: 1024
-  terrain_type: plane
-  terrain_generator: null
-  usd_path: null
-  env_spacing: 4.0
-  visual_material:
-    func: isaaclab.sim.spawners.materials.visual_materials:spawn_preview_surface
-    diffuse_color: !!python/tuple
-    - 0.0
-    - 0.0
-    - 0.0
-    emissive_color: !!python/tuple
-    - 0.0
-    - 0.0
-    - 0.0
-    roughness: 0.5
-    metallic: 0.0
-    opacity: 1.0
-  physics_material:
-    func: isaaclab.sim.spawners.materials.physics_materials:spawn_rigid_body_material
-    static_friction: 1.0
-    dynamic_friction: 1.0
-    restitution: 0.0
-    friction_combine_mode: average
-    restitution_combine_mode: average
-    compliant_contact_stiffness: 0.0
-    compliant_contact_damping: 0.0
-  max_init_terrain_level: null
-  debug_vis: false
-cmd_lin_range:
-- -1.0
-- 1.0
-cmd_ang_range:
-- -1.0
-- 1.0
-termination_height: 0.3
-termination_roll: 1.0
-termination_pitch: 1.0
-reward_lin_vel_w: 2.0
-reward_ang_vel_w: 2.0
-reward_orientation_w: 0.5
-reward_torque_reg_w: 0.01
-reward_action_rate_w: 0.05
-reward_alive_w: 0.2
-reward_failure_penalty: -100.0
-lin_vel_sigma: 0.5
-ang_vel_sigma: 0.5
-orientation_sigma: 0.5
-action_rate_sigma: 0.2
diff --git a/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py b/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py
index c4ec117..e6e1fd4 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env.py
@@ -90,7 +90,7 @@ class DodoVelocityTrackingEnv(LocomotionEnv):
 
         self.body_pos_w = self.robot.data.body_pos_w
         feet_z = self.body_pos_w[:, self._feet_body_ids, 2]
-        feet_in_air = feet_z > 0.05
+        feet_in_air = feet_z > 0.10
         self.num_feet_in_air = torch.sum(feet_in_air.float(), dim=1)
 
     def _get_observations(self) -> dict:
@@ -140,6 +140,12 @@ class DodoVelocityTrackingEnv(LocomotionEnv):
         action_diff_sq = torch.mean((self.actions - self.prev_actions) ** 2, dim=1)
         r_action_rate = self.cfg.reward_action_rate_w * torch.exp(-action_diff_sq / (2 * self.cfg.action_rate_sigma ** 2))
 
+        # reward of knee angle
+        left_knee_angle = self.motor_pos[:, 2]
+        right_knee_angle = self.motor_pos[:, 5]
+        knee_angle_err_sq = (left_knee_angle - self.cfg.knee_angle_goal) ** 2 + (right_knee_angle - self.cfg.knee_angle_goal) ** 2
+        r_knee_angle = self.cfg.reward_knee_angle * torch.exp(-knee_angle_err_sq / (2 * self.cfg.knee_angle_sigma ** 2))
+
         # alive 
         r_alive = torch.ones(self.num_envs, device=self.sim.device) * self.cfg.reward_alive_w
 
@@ -151,7 +157,11 @@ class DodoVelocityTrackingEnv(LocomotionEnv):
         p_fail = failure.float() * self.cfg.reward_failure_penalty
 
         # foot in air reward
-        r_foot_in_air = self.cfg.reward_foot_in_air * (self.num_feet_in_air / len(self._feet_body_ids))
+        # reward if exactly one foot is in the air; penalize otherwise (both on ground or both in air)
+        one_foot_mask = (self.num_feet_in_air == 1)
+        pos_val = torch.full_like(self.num_feet_in_air, self.cfg.reward_foot_in_air)
+        neg_val = torch.full_like(self.num_feet_in_air, -self.cfg.reward_foot_in_air)
+        r_foot_in_air = torch.where(one_foot_mask, pos_val, neg_val)
 
         # total reward
         total_reward = r_lin + r_ang + r_orient + r_torque + r_action_rate + r_alive + p_fail + r_foot_in_air
diff --git a/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py b/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py
index 6396926..4272f17 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/direct/dodo_velocity_tracking/dodo_velocity_tracking_env_cfg.py
@@ -60,11 +60,14 @@ class DodoVelocityTrackingEnvCfg(DirectRLEnvCfg):
     ]
 
     foot_link_names: list[str] = ["left_link_4", "right_link_4"]
+    knee_link_names: list[str] = ["left_link_3", "right_link_3"]
 
     # task
     cmd_lin_range = [1.5, 2.5]  # m/s
     cmd_ang_range = [-2.0, 2.0]  # rad/s
 
+    knee_angle_goal = 1.0  # rad
+
     termination_height = 0.3  # m
     termination_roll = 1.0  # rad
     termination_pitch = 1.0  # rad
@@ -76,6 +79,7 @@ class DodoVelocityTrackingEnvCfg(DirectRLEnvCfg):
     reward_torque_reg_w = 0.01
     reward_action_rate_w = 0.05
     reward_alive_w = 0.2
+    reward_knee_angle = 0.8
     reward_foot_in_air = 2.0
     reward_failure_penalty = -20.0
 
@@ -84,3 +88,4 @@ class DodoVelocityTrackingEnvCfg(DirectRLEnvCfg):
     ang_vel_sigma = 0.5
     orientation_sigma = 0.5
     action_rate_sigma = 0.2
+    knee_angle_sigma = 0.5
diff --git a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
index c3c8b0c..8978dc5 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
@@ -19,8 +19,8 @@ from isaaclab.sensors import ContactSensorCfg
 from isaaclab.terrains import TerrainImporterCfg
 from isaaclab.utils import configclass
 
-import dodo_manage.tasks.manager_based.dodo_manage.mdp as mdp
-from isaac_dodo.isaac_dodo.assets.robots.dodo import DODO_CFG
+import isaac_dodo.tasks.manager_based.dodo_manage.mdp as mdp
+from isaac_dodo.assets.robots.dodo import DODO_CFG
 
 ##
 # Scene definition
diff --git a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py
index 8efe831..e796861 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py
@@ -17,7 +17,7 @@ from isaaclab.managers import ManagerTermBase, RewardTermCfg, SceneEntityCfg
 from isaaclab.utils.math import quat_apply_inverse, yaw_quat
 
 
-import dodo_manage.tasks.manager_based.dodo_manage.mdp.observations as obs
+import isaac_dodo.tasks.manager_based.dodo_manage.mdp.observations as obs
 
 if TYPE_CHECKING:
     from isaaclab.envs import ManagerBasedRLEnv