--- git status ---
On branch joystick-tracking-trash
Your branch is up to date with 'origin/joystick-tracking-trash'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt
	modified:   logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt
	modified:   logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd
	modified:   source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd
	modified:   source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
	modified:   source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	logs/rsl_rl/dodo_manage_joystick_tracking_ppo/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_14-18-24/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_14-40-21/
	logs/rsl_rl/dodo_managed_ppo/2026-01-29_14-55-25/
	logs/rsl_rl/dodo_velocity_tracking_ppo/2026-01-29_14-05-14/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt b/logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt
index bdde2a5..ab613f9 100644
Binary files a/logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt and b/logs/rsl_rl/dodo_managed_ppo/2026-01-15_18-41-11/exported/policy.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt
index 6fd3518..c7e993b 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_250.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt
index 6e413b2..29c589c 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_300.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt
index 9520486..87f3f82 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_350.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt
index 098d581..615005b 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_400.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt
index 84a7351..fdda736 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_450.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt
index a4fd06f..b82bc1e 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_50.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt
index 343307d..b43e24c 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_500.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt
index 8a4485d..3e6a3f3 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_550.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt
index 30bce28..dd20a86 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_600.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt
index dc0e874..0017616 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_650.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt
index 3af0b31..88ac861 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_700.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt
index d16f272..6775546 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_750.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt
index d49b062..022cfc1 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_800.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt
index 4b6a6b4..1931741 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_850.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt
index c27f3d9..13d287b 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_900.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt
index 018ab73..d60cad2 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_950.pt differ
diff --git a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt
index 0efc4e8..57c6eba 100644
Binary files a/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt and b/logs/rsl_rl/dodo_velocity_tracking_ppo/2025-11-20_18-17-41/model_999.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt
index 4c96b61..cc5b398 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/exported/policy.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt
index a235fde..3711c83 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_0.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt
index f7fe214..973c401 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_100.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt
index 36d574d..567f585 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_200.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt
index 5c2f48d..380578f 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_300.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt
index 189d172..f52b94b 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_400.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt
index 79380b9..3e92cf7 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_500.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt
index a9dbe7e..7e42eb8 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_600.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt
index e4e0ca5..0e44dab 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_700.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt
index 1c0a07a..5d58a5c 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_800.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt
index 12957f7..600e3be 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_900.pt differ
diff --git a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt
index 35099b6..4e7860a 100644
Binary files a/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt and b/logs/rsl_rl/model_saved/2026-01-15_16-29-37/model_999.pt differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd
index a7adf67..838260c 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_base.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd
index 2697045..f7182a7 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_physics.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd
index 2af19ea..dd8225f 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_robot.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd
index 00e94a9..40c4936 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/configuration/dodobot_v3_simple_sensor.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd
index ae3974d..3e63b9d 100644
Binary files a/source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd and b/source/isaac_dodo/isaac_dodo/assets/robots/dodo/dodobot_v3.usd differ
diff --git a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
index 6ae6a3a..68d32a4 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/dodo_manage_env_cfg.py
@@ -197,8 +197,8 @@ class EventCfg:
         func=mdp.reset_joints_by_offset,
         mode="reset",
         params={
-            "position_range": (-0.2, 0.2),
-            "velocity_range": (-0.1, 0.1),
+            "position_range": (-0.05, 0.05),
+            "velocity_range": (-0.05, 0.05),
         },
     )
 
@@ -233,12 +233,15 @@ class RewardsCfg:
         params={"command_name": "base_velocity", "std": 0.5},
     )
 
+    yaw_rate = RewTerm(func=mdp.yaw_rate_l2, weight=-0.2)
+    lin_vel_y = RewTerm(func=mdp.lin_vel_y_l2, weight=-0.2)
+
     feet_air_time = RewTerm(
         func=mdp.feet_air_time_positive_biped_snesor,
         weight=0.3,
         params={
             "command_name": "base_velocity",
-            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
+            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=["left_link_4", "right_link_4"]),
             "threshold": 0.4,
         },
     )
@@ -250,10 +253,10 @@ class RewardsCfg:
         
     feet_slide = RewTerm(
         func=mdp.feet_slide,
-        weight=-2.0,
+        weight=-0.2,
         params={
-            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=".*_link_4"),
-            "asset_cfg": SceneEntityCfg("robot", body_names=".*_link_4"),
+            "sensor_cfg": SceneEntityCfg("contact_forces", body_names=["left_link_4", "right_link_4"]),
+            "asset_cfg": SceneEntityCfg("robot", body_names=["left_link_4", "right_link_4"]),
         },
     )
 
diff --git a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py
index 3971ccb..2046d65 100644
--- a/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py
+++ b/source/isaac_dodo/isaac_dodo/tasks/manager_based/dodo_manage/mdp/rewards.py
@@ -214,7 +214,15 @@ def feet_lateral_separation_reward(
 
     return r * stance_any.float()
 
-# 
+# 	yaw 角速度惩罚（让它别自转）
+def yaw_rate_l2(env, asset_cfg=SceneEntityCfg("robot")):
+    asset = env.scene[asset_cfg.name]
+    return asset.data.root_ang_vel_w[:, 2] ** 2
+
+# 	侧向速度惩罚（让它别横着走）
+def lin_vel_y_l2(env, asset_cfg=SceneEntityCfg("robot")):
+    asset = env.scene[asset_cfg.name]
+    return asset.data.root_lin_vel_w[:, 1] ** 2
 
 
 # Unused reward terms (currently disabled in env cfg)